### PR TITLE
Google analytics events

### PR DESCRIPTION
**Pull request by annyhe**
add tracking for 
- search command
- outgoing links in the modal
- onclick modal
- check/uncheck the "Show All" box

### CODE REVIEW

**Comment by annyhe** [comment]
This lens is deployed to https://refocus-sandbox.internal.salesforce.com/perspectives/demo_with_google_analytics.
See the events being logged in the chrome networks tab, whenever check/uncheck the show all box, do a find, ... 
screen shot 2017-08-04 at 2 16 25 pm

File: MultiTable/src/main.js [hunk 1]
```diff
@@ -45,6 +45,13 @@ let subjectModal;
 let aspectModal;
 let urlParams = {};
 
+// for (ctrl + f) or (cmd + f)
```
Comment (iamigo) [inline MultiTable/src/main.js]: The PR should also include the update to the version and release notes in the lens.json, plus build the lens to generate the zip file, plus include the new zip file in the PR

File: MultiTable/src/main.js [hunk 2]
```diff
@@ -232,13 +257,21 @@ function bindContentToModal(modal, modalTemplate, context, content) {
   const str = modalTemplate(context);
   modal.innerHTML = '';
   modal.insertAdjacentHTML('beforeend', str);
+
+  // add event handlers to detect clickingo of links
```
Comment (iamigo) [inline MultiTable/src/main.js]: "clickingo"?

File: MultiTable/lens.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
   "helpEmail": "igoldstein@salesforce.com",
   "helpUrl": "https://github.com/salesforce/refocus-lens-multitable",
   "license": "TBD",
-  "version": "1.0.19",
+  "version": "1.0.20",
```
Comment (iamigo) [inline MultiTable/lens.json]: need to include the new zip in the PR, too

File: MultiTable/src/main.js [hunk 4]
```diff
@@ -45,6 +45,13 @@ let subjectModal;
 let aspectModal;
 let urlParams = {};
 
+// for (ctrl + f) or (cmd + f)
+window.onkeydown = function(e) {
+  if (ga && e.keyCode == 70 && (e.ctrlKey || e.metaKey)) {
+    ga('send', 'event', 'Lens - Multitable', 'Search Command');
```
Comment (iamigo) [inline MultiTable/src/main.js]: Maybe call this "Browser Search Command"?

File: MultiTable/src/main.js [hunk 5]
```diff
@@ -116,6 +135,12 @@ function onHierarchyLoad(evt) {
   if (showAll) data.reset(showAll);
 
   $('#toggle-show-all').change((evt) => {
+    if (evt.target.checked == true) {
```
Comment (iamigo) [inline MultiTable/src/main.js]: trackClick(evt.target.checked ? 'check' : 'uncheck', 'Show All');

File: MultiTable/src/main.js [hunk 6]
```diff
@@ -232,13 +257,21 @@ function bindContentToModal(modal, modalTemplate, context, content) {
...
+    });
+  });
 } // bindContentToModal
 
 function setSampleListeners(subjectGroup) {
   const samples = document.getElementById(subjectGroup.key)
     .querySelectorAll('.sample');
   samples.forEach((sample) => {
     sample.addEventListener('click', (evt) => {
+      trackClick('click sample', evt.target.id);
```
Comment (iamigo) [inline MultiTable/src/main.js]: consistent use of title case or all lowercase across all trackClick names being assigned

File: MultiTable/src/main.js [hunk 7]
```diff
@@ -304,7 +338,7 @@ function setAspectListeners(subjectGroup) {
   });
 } // setAspectListeners
 
-handlebars.registerHelper('flattenRange', function (range) {
+handlebars.registerHelper('flattenRange', function(range) {
```
Comment (iamigo) [inline MultiTable/src/main.js]: (range) => { ... }

File: MultiTable/src/main.js (hunk 6 above)
Comment (annyhe) [inline MultiTable/src/main.js]: yep chose to go with lower case.

File: MultiTable/src/main.js [hunk 8]
```diff
@@ -30,7 +30,8 @@ const template = {
 const RealtimeChangeHandler = require('./RealtimeChangeHandler');
 const SubjectGroups = require('./SubjectGroups');
 
-const EVENT_CATEGORY = 'Lens - Multitable';
+const VERSION = require('../lens.json').version;
```
Comment (iamigo) [inline MultiTable/src/main.js]: Must build and update the zip in the PR too

Comment (annyhe) [inline MultiTable/src/main.js]: see above comment for the updated zip. 
Here's what the category looks like, with lens version
screen shot 2017-08-07 at 10 36 28 am

Comment (iamigo) [inline MultiTable/src/main.js]: must include the newest generated zip after these latest changes

**Pull request merged**